go/cbindgen.toml

```diff
@@ -0,0 +1,7 @@
+language = "C"
```

Comment:
why not use uniffi and generate the whole go code?

https://github.com/NordSecurity/uniffi-bindgen-go

Reply:
We ran into some issues using uniffi for Golang. The examples require adding uniffi as a dependency and build dependency in order to generate the bindings on the Rust side. There's likely still a way around this, but it'll require more time to investigate and the team would rather move forward with Protobuf for now and do this investigation later, especially because of the new direction we're heading in focusing on the Java client.

The generated bindings also create a separate Go file that handles all the data marshaling between Go and Rust and hides cgo from the Go wrapper code. While this makes it easier to pass data back and forth, it also raises questions about performance. We'd need to benchmark the wrapper with the bindings to make sure that the overhead isn't problematic.

I've created a ticket in our fork of the repo to investigate this later.

Reply:
then please add a TODO in the code here and in the Go bindings, to make it clear that this is a temporary crutch.